Mathlib/Algebra/Lie/Nilpotent.lean

```diff
@@ -68,6 +70,11 @@ theorem lcs_succ : N.lcs (k + 1) = ⁅(⊤ : LieIdeal R L), N.lcs k⁆ :=
   Function.iterate_succ_apply' (fun N' => ⁅⊤, N'⁆) k N
 #align lie_submodule.lcs_succ LieSubmodule.lcs_succ
 
+lemma lcs_sup {N₁ N₂ : LieSubmodule R L M} {k : ℕ} :
+    (N₁ ⊔ N₂).lcs k = N₁.lcs k ⊔ N₂.lcs k := by
```

Comment:
Is the `iSup` version true as well?

Reply:
Yes but might be a bit more work, I'll see.

Reply:
Yeah this is a bit of work because it needs the missing lemma:
```lean

@[simp]
theorem lie_iSup {ι : Type*} {N : ι → LieSubmodule R L M} :
    ⁅I, ⨆ i, N i⁆ = ⨆ i, ⁅I, N i⁆ := by
  refine le_antisymm ?_ (iSup_le_iff.mpr fun i ↦ mono_lie_right _ _ _ (le_iSup N i))
  rw [← coeSubmodule_le_coeSubmodule, lieIdeal_oper_eq_linear_span, Submodule.span_le]
  rintro - ⟨⟨x, hx⟩, ⟨m, hm⟩, rfl⟩
  simp only [SetLike.mem_coe, mem_coeSubmodule]
  -- etc but we now run into missing API for `iSup`s of LieSubmodules
  sorry
```
I propose not adding the `iSup` version of this lemma because:

1. I currently have no use for it,
2. In future work, I want to change the `CompleteLattice` definition on `LieSubmodule` to be definitionally equal to that of `Submodule`: this will tidy things up a lot and make adding missing proofs like the above much easier.

Reply:
The missing API for `iSup`s of `LieSubmodule`s is added in #7608